### PR TITLE
src/test/common/test_hobject: remove constexpr

### PR DESCRIPTION
**Pull request by rkhudov**
Remove `constexpr` from `src/test/common/test_hobject.cc`. It creates 2 issues when running `ninja tests` command.
Issues:
1. error: call to non-'constexpr' function 'bool test_hobject_fmt_t::is_max() const'
```
 /ceph-src/src/test/common/test_hobject.cc: In member function 'constexpr auto test_hobject_fmt_t::operator<=>(const test_hobject_fmt_t&) const':
/ceph-src/src/test/common/test_hobject.cc:103:22: error: call to non-'constexpr' function 'bool test_hobject_fmt_t::is_max() const'
103 |     auto cmp = is_max() <=> rhs.is_max();
|                ~~~~~~^~
/ceph-src/src/test/common/test_hobject.cc:94:8: note: 'bool test_hobject_fmt_t::is_max() const' declared here
94 |   bool is_max() const { return max; }
```
2.  error: temporary of non-literal type 'const test_hobject_fmt_t' in a constant expression
```
/ceph-src/src/test/common/test_hobject.cc:74:8: note: 'test_hobject_fmt_t' is not literal because:
74 | struct test_hobject_fmt_t : public hobject_t {
 |        ^~~~~~~~~~~~~~~~~~
/ceph-src/src/test/common/test_hobject.cc:74:8: note:   'test_hobject_fmt_t' does not have 'constexpr' destructor
/ceph-src/src/test/common/test_hobject.cc: In member function 'virtual void HObject_fmt_random_Test::TestBody()':
```

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [x] Very recent bug; references [commit](https://github.com/ceph/ceph/commit/94f1cadda240292086eae7d1e484503005a6ceb1) where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by Matan-B — APPROVED** [review]

**Comment by ronen-fr** [comment]
I am not seeing the compiler errors above in the environment I've tested. I have no objection to this fix, but plan to
possibly reintroduce the constexpr as part of a larger pending change to this file.

File: src/test/common/test_hobject.cc [hunk 1]
```diff
@@ -201,7 +201,7 @@ namespace fmt {
 template <>
 struct formatter<test_hobject_fmt_t> {
 
-  constexpr auto parse(format_parse_context& ctx) { return ctx.begin(); }
+  auto parse(format_parse_context& ctx) { return ctx.begin(); }
```
Comment (ronen-fr) [inline src/test/common/test_hobject.cc:204]: why is this change required?

**Review by ronen-fr — APPROVED** [review]
Please see my note on line 204 before continuing with the PR.
LGTM (for now) otherwise

File: src/test/common/test_hobject.cc (hunk 1 above)
Comment (rkhudov) [reply src/test/common/test_hobject.cc:204]: Because without this change I have compilation error
Which gcc version your environment has?

Comment (rkhudov) [reply src/test/common/test_hobject.cc:204]: I got this compilation error on Ubuntu and Fedora machines

Comment (ronen-fr) [reply src/test/common/test_hobject.cc:204]: Was Fedora 37 (don't remember what the version was). I've just (last night) upgraded to 39 & gcc 13.2.1, and
will test there . On Clang - compiled OK on both platforms (and in the multiple  CI environments).

Anyway - can be merged now (I do not see a reason for a Teuthology test run for a single test change).

Comment (rkhudov) [reply src/test/common/test_hobject.cc:204]: Try to test on gcc11, because I think this is the default one for ceph
